src/image.rs

```diff
@@ -0,0 +1,200 @@
+        })
+    }
+
+    fn user(&self) -> &user::User {
+        &self.user
+    }
+
+    fn distribution(&self) -> &str {
+        &self.distribution
+    }
+
+    fn version(&self) -> &str {
+        &self.version
+    }
+
+    /// 任意のディストリビューション、バージョンのrootfsやhash fileを入れるpathを返却
+    ///
+    /// # Example
+    /// ```~/.injesh/images/alpine/3.15```
+    fn image_base_path(&self) -> &Path {
+        self.image_base_path.as_path()
+    }
+
+    /// 任意のディストリビューション、バージョンのhash fileへのpathを返却
+    ///
+    /// # Example
+    /// ```~/.injesh/images/alpine/3.15/rootfs.tar.xz.asc```
+    fn rootfs_hash_path(&self) -> PathBuf {
+        PathBuf::from(&format!(
+            "{}/{}",
+            self.image_base_path().display(),
+            ROOTFS_HASH_FILE
+        ))
+    }
+
+    /// 任意のディストリビューション、バージョンのダウンロードされるrootfs.tar.xzへのpathを返却
+    /// # Example
+    /// ```~/.injesh/images/alpine/3.15/rootfs.tar.xz```
+    fn downloaded_rootfs_path(&self) -> PathBuf {
+        PathBuf::from(&format!(
+            "{}/{}",
+            self.image_base_path().display(),
+            ROOTFS_FILE
+        ))
+    }
+
+    /// 任意のディストリビューション、バージョンのrootfsへのpathを返却
+    ///
+    /// # Example
+    /// ```~/.injesh/images/alpine/3.15/rootfs```
+    fn rootfs_path(&self) -> PathBuf {
+        PathBuf::from(&format!("{}/{}", self.image_base_path().display(), ROOTFS))
+    }
+
+    /// ローカルにrootfsイメージがあるかどうか調べる
+    ///
+    /// # Example
+    /// ```ignore
+    /// image.search_image();
+    /// ```
+    pub fn search_image(&self) -> Result<(), Box<dyn std::error::Error>> {
+        if self.rootfs_hash_path().exists() {
+            return Ok(());
+        }
+        Err(Error::ImageNotFound)?
+    }
+
+    /// ローカルにあるイメージが最新のものかどうかを調べる
+    ///
+    /// # Example
+    /// ```ignore
+    /// image.image_is_newes();
+    /// ```
+    pub fn check_rootfs_newest(&self) -> Result<bool, Box<dyn std::error::Error>> {
+        self.downloader
+            .check_rootfs_newest(&self.rootfs_hash_path())
+    }
+
+    /// rootfsイメージをダウンロードする
+    /// 既にrootfsが存在する場合、削除してからダウンロードする
+    /// そのため存在するかどうか、最新かどうかを確認してから
+    /// 呼び出す必要がある
+    ///
+    /// # Example
+    /// ```ignore
+    /// image.download_image();
+    /// ```
+    pub fn download_image(&self) -> Result<(), Box<dyn std::error::Error>> {
+        // ローカルにあるrootfsが最新のものかどうかを確認
+        if Path::new(&self.rootfs_hash_path()).exists() {
+            match self.check_rootfs_newest() {
+                Ok(newest_flg) => match newest_flg {
+                    true => return Ok(()),
+                    false => { /* do nothing */ }
+                },
+                Err(_) => { /* do nothing */ }
+            }
+        }
+
+        // rootfsディレクトリがないならば作る
+        // ある場合はrootfsディレクトリ以下を削除する
+        if !Path::new(&self.rootfs_path()).exists() {
+            fs::create_dir_all(&self.rootfs_path())?;
+        } else {
+            fs::remove_dir_all(&self.rootfs_path())?;
+        }
```

Comment:
コメントあるからめっちゃわかるんだけど，コード読むと引っかかるので:
```suggestion
        // rootfsディレクトリがある場合は最新ではないのでrootfsディレクトリ以下を削除する
        // ないならば作る
        if Path::new(&self.rootfs_path()).exists() {
            fs::remove_dir_all(&self.rootfs_path())?;
        } else {
            fs::create_dir_all(&self.rootfs_path())?;
        }
```